src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs

```diff
@@ -328,55 +328,50 @@ public override string ToString()
 
     public class NodeBuilder : IDisposable
     {
-        public static NodeBuilder Create([CallerMemberNameAttribute] string caller = null, string version = "0.13.1")
+        /// <summary>
+        /// Deletes test folders. Stops "bitcoind" if required.
+        /// </summary>
+        /// <param name="folder">The folder to remove.</param>
+        /// <returns>Returns true if the folder was successfully removed and false otherwise.</returns>
+        public static bool CleanupTestFolder(string folder)
         {
-            version = version ?? "0.13.1";
-            if (!Directory.Exists("TestData"))
-                Directory.CreateDirectory("TestData");
-            var path = EnsureDownloaded(version);
-            caller = Path.Combine("TestData", caller);
-            bool retryDelete = true;
-            try
-            {
-                Directory.Delete(caller, true);
-                retryDelete = false;
-            }
-            catch (DirectoryNotFoundException)
-            {
-                retryDelete = false;
-            }
-            catch (UnauthorizedAccessException)
-            {
-            }
-            catch (IOException)
-            {
-            }
-            if (retryDelete)
+            bool tryKill = true;
+            while (true)
             {
+                try
+                {
+                    Directory.Delete(folder, true);
+                    return true;
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return true;
+                }
+                catch (Exception)
+                {
+                }
+                if (!tryKill)
+                    return false;
+                tryKill = false;
```

Comment:
Is the purpose of tryKill to try deleting one more time?
If that's the case, then having a while(true) + a flag (tryKill) seems a bit overengineered if all we want to do is try to delete something twice. In this case, a for loop with a retryCount of 2 seems good enough.